code/modules/integrated_electronics/core/integrated_circuit.dm

```diff
@@ -0,0 +1,355 @@
+			switch(j)
+				if(1)
+					io = get_pin_ref(IC_INPUT, i)
+					if(io)
+						words += "<b><a href=?src=\ref[src];pin_name=1;pin=\ref[io]>[io.display_pin_type()] [io.name]</a> <a href=?src=\ref[src];pin_data=1;pin=\ref[io]>[io.display_data(io.data)]</a></b><br>"
+						if(io.linked.len)
+							for(var/datum/integrated_io/linked in io.linked)
+								words += "<a href=?src=\ref[src];pin_unwire=1;pin=\ref[io];link=\ref[linked]>[linked.name]</a> \
+								@ <a href=?src=\ref[linked.holder];examine=1;>[linked.holder.displayed_name]</a><br>"
+
+						if(outputs.len > inputs.len)
+							height = 1
+				if(2)
+					if(i == 1)
+						words += "[src.displayed_name]<br>[src.name != src.displayed_name ? "([src.name])":""]<hr>[src.desc]"
+						height = row_height
+					else
+						continue
+				if(3)
+					io = get_pin_ref(IC_OUTPUT, i)
+					if(io)
+						words += "<b><a href=?src=\ref[src];pin_name=1;pin=\ref[io]>[io.display_pin_type()] [io.name]</a> <a href=?src=\ref[src];pin_data=1;pin=\ref[io]>[io.display_data(io.data)]</a></b><br>"
+						if(io.linked.len)
+							for(var/datum/integrated_io/linked in io.linked)
+								words += "<a href=?src=\ref[src];pin_unwire=1;pin=\ref[io];link=\ref[linked]>[linked.name]</a> \
+								@ <a href=?src=\ref[linked.holder];examine=1;>[linked.holder.displayed_name]</a><br>"
+
+						if(inputs.len > outputs.len)
+							height = 1
+			HTML += "<td align='center' rowspan='[height]'>[jointext(words, null)]</td>"
+		HTML += "</tr>"
+
+	for(var/activator in activators)
+		var/datum/integrated_io/io = activator
+		var/words = list(
+			"<b><a href=?src=\ref[src];pin_name=1;pin=\ref[io]><span class='bad'>[io.name]</span></a> <a href=?src=\ref[src];pin_data=1;pin=\ref[io]><span class='bad'>[io.data?"\<PULSE OUT\>":"\<PULSE IN\>"]</span></a></b><br>"
+		)
+
+		if(io.linked.len)
+			for(var/datum/integrated_io/linked in io.linked)
+				words += "<a href=?src=\ref[src];pin_unwire=1;pin=\ref[io];link=\ref[linked]><span class='bad'>[linked.name]</span></a> \
+				@ <a href=?src=\ref[linked.holder];examine=1;><span class='bad'>[linked.holder.displayed_name]</span></a><br>"
+
+		HTML += "<tr>"
+		HTML += "<td colspan='3' align='center'>[jointext(words, null)]</td>"
+		HTML += "</tr>"
+
+	HTML += "</table>"
+	HTML += "</div>"
+
+//	HTML += "<br><font color='33CC33'>Meta Variables;</font>" // If more meta vars get introduced, uncomment this.
+//	HTML += "<br>"
+
+	HTML += "<br><span class='highlight'>Complexity: [complexity]</span>"
+	if(power_draw_idle)
+		HTML += "<br><span class='highlight'>Power Draw: [power_draw_idle] W (Idle)</span>"
+	if(power_draw_per_use)
+		HTML += "<br><span class='highlight'>Power Draw: [power_draw_per_use] W (Active)</span>" // Borgcode says that powercells' checked_use() takes joules as input.
+	HTML += "<br><span class='highlight'>[extended_desc]</span>"
+
+	var/datum/browser/B = new(user, assembly ? "assembly-\ref[assembly]" : "circuit-\ref[src]", name, window_width, window_height)
```

Comment:
Could probably replace `name` with `displayed_name ? [displayed_name] : [name]` or something similar, to help with organisation. As-is, the display name doesn't show up at the top when managing an individual circuit, even though it shows up in the middle: http://puu.sh/xxXKU/b46e32b690.png